chains/nomad-ethereum/src/gelato/types.rs

```diff
@@ -0,0 +1,167 @@
+use super::utils::CHAIN_ID_TO_FORWARDER;
+use ethers::abi::{self, Token};
+use ethers::types::{transaction::eip712::*, Address};
+use ethers::utils::hex::FromHexError;
+use ethers::utils::keccak256;
+use gelato_relay::ForwardRequest;
+use std::str::FromStr;
+
+const FORWARD_REQUEST_TYPE: &str = "ForwardRequest(uint256 chainId,address target,bytes data,address feeToken,uint256 paymentType,uint256 maxFee,uint256 gas,address sponsor,uint256 sponsorChainId,uint256 nonce,bool enforceSponsorNonce,bool enforceSponsorNonceOrdering)";
+
+/// Unsigned forward request
+#[derive(Debug, Clone)]
+pub struct UnsignedFowardRequest {
+    /// Chain id
+    pub chain_id: usize,
+    /// Target contract
+    pub target: String,
+    /// Encoded tx data
+    pub data: String,
+    /// Fee token
+    pub fee_token: String,
+    /// Payment method
+    pub payment_type: usize, // 1 = gas tank
```

Comment:
This should be encoded in an `enum` or `const`

Reply:
enum: Technically yes but we need this to be usize because it gets encoded/signed as number as part of EIP 712 standard. More straightforward to keep the type the same as what ultimately ends up getting signed

const: we could technically have this be something other than `1` if we use diff payment type one day so don't want a const

Reply:
An `enum` can be an int, but it's probably not worth the casting, etc.

I meant `const FOO_BAR_GASTANK: usize = 1` so we know what it is apart from a comment. I wouldn't worry about it now though.